Create send tx guide - Closes #1258, #1428 (#1443)
LiskHQ/lisk-docs: docs/modules/ROOT/pages/integrate-blockchain/create-send-transactions.adoc

  ],
  id: <Buffer 17 a0 90 44 15 f7 ad 9a 0e 30 5a e8 a4 34 53 24 3a 84 e2 af e1 76 08 3a 0a b3 3f b1 2e 3b bc 25>
}
----

=== JSON compatible object

A representation of the decoded transaction in a JSON compatible format.

Used for example for displaying the details of a transaction in the UI to the user.

[source,js]
----
{
  moduleID: 2,
  assetID: 0,
  nonce: '4',
  fee: '1000000',
  senderPublicKey: '5133af7944acf5278b0310a11c06134f80ab4546d77d1b0e027c8430a7d2bb92',
  signatures: [
    '98a9ee2cde8354d014cfe6367d430be2713e102f37d92ab91f03db780407e5bf6d818a45c21c9f5518638dfc3c5365fc2d497b928e0b9d6337988df46a663a02'
  ],
  asset: {
    amount: '800000000',
    recipientAddress: '9bd82e637d306533b1e1ad66e19ca0047faa1a6a',
    data: 'Happy birthday!'
  },
  id: '17a0904415f7ad9a0e305ae8a43453243a84e2afe176083a0ab33fb12e3bbc25'
}
----

== How to create a transaction

[TIP]
====
For convenience, it is recommended to use the API client or the CLI to communicate to a node for creating and signing transactions.

> In case a higher security is desired, it is possible to create the transaction in a complete offline environment.

Reviewer: ```suggestion
In the case whereby a higher level of security is desired, it is possible to create the transaction in a completely stand-alone offline environment.
```

Please check out the guide xref:{url_integrate_tx_offline}[] for more information.
====

.Transaction properties
[NOTE]
====

> The properties of a transaction are described in detail in the Lisk Protocol at the xref:{url_protocol_txs}[Transactions] page.

Reviewer: ```suggestion
The properties of a transaction are described in detail in the Lisk Protocol on the xref:{url_protocol_txs}[Transactions] page.
```

====

=== via API client

> The API client allows connecting to a particular node API, and enables to get and post data to the node.

Reviewer: ```suggestion
The API client allows connecting to a particular node API, and enables the user to get and post data to the node.
```

----
$ lisk-core transaction:create 2 0 100000000 --json --pretty
? Please enter: amount:  1000000000
? Please enter: recipientAddress:  ab0041a7d3f7b2c290b5b834d46bdc7b7eb85815
? Please enter: data:  send tokens
? Please enter passphrase:  [hidden]
? Please re-enter passphrase:  [hidden]
{
  "transaction": "0802100018022080c2d72f2a20e03c09bdc8c023d94cf66a5d352e6258380210d97d545abbf75668ea3736e3123229088094ebdc031214ab0041a7d3f7b2c290b5b834d46bdc7b7eb858151a0b73656e6420746f6b656e733a40faa2626d7306506b1999f48aa2f4b1ffdee01e641fa76d37a9d1d6fd8c225a81065c856ea625c52d138a7e3ba86b62913dc8e5aef8b5e307641ab66e0277a60b"
}
{
  "transaction": {
    "moduleID": 2,
    "assetID": 0,
    "nonce": "2",
    "fee": "100000000",
    "senderPublicKey": "e03c09bdc8c023d94cf66a5d352e6258380210d97d545abbf75668ea3736e312",
    "signatures": [
      "faa2626d7306506b1999f48aa2f4b1ffdee01e641fa76d37a9d1d6fd8c225a81065c856ea625c52d138a7e3ba86b62913dc8e5aef8b5e307641ab66e0277a60b"
    ],
    "asset": {
      "amount": "1000000000",
      "recipientAddress": "ab0041a7d3f7b2c290b5b834d46bdc7b7eb85815",
      "data": "send tokens"
    }
  }
}
----
=====
====

== How to send the transaction

[TIP]
====

> In case it is desired to have the transaction in a different format before sending, there are functions to convert the transaction between the different formats conveniently, see xref:{url_integrate_decoding}[] for more information.

Reviewer: ```suggestion
In case it is desired to have the transaction in a different format before sending, there are functions available to conveniently convert the transaction between the different formats, see xref:{url_integrate_decoding}[] for more information.
```

[source,js]
----
// 1. Require the dependencies
const io = require('socket.io-client'); // The socket.io client
const jsome = require('jsome'); // Prettifies the JSON output

jsome.params.colored = true;

// Use local Service node
const WS_RPC_ENDPOINT = 'ws://localhost:9901/rpc-v2';
//Use public Service node
//const WS_RPC_ENDPOINT = "wss://service.lisk.com/rpc-v2";

// 2. Connect to Lisk Service via WebSockets
const socket = io(WS_RPC_ENDPOINT, {
  forceNew: true,
  transports: ['websocket']
});

// 3. Emit the remote procedure call
socket.emit('request', {
  jsonrpc: '2.0',
  method: 'post.transactions',
  payload: {"transaction":"08021000180d2080c2d72f2a200fe9a3f1a21b5530f27f87a414b549e79a940bf24fdf2b2f05e7f22aeeecc86a32270880c2d72f12144fd8cc4e27a3489b57ed986efe3d327d3de40d921a0a73656e6420746f6b656e3a4069242925e0e377906364fe6c2eed67f419dfc1a757f73e848ff2f1ff97477f90263487d20aedf538edffe2ce5b3e7601a8528e5cd63845272e9d79c294a6590a"}
},
  answer => {
    // console.log(answer);
    jsome(answer);
    process.exit(0);
});
----

[TIP]
====
For more information, check out the xref:{url_api_service_rpc}[] reference.
====

=== With the application / node

==== Via the application CLI

> Any running node with enabled API can be used to send transaction object, see xref:{url_sdk_cli}[] and xref:{url_core_cli}[].

Reviewer: ```suggestion
Any running node with an enabled API can be used to send a transaction object, see xref:{url_sdk_cli}[] and xref:{url_core_cli}[].
```

An example for sending a transfer transaction with the Lisk Core CLI is displayed below:

[source,bash]
----
$ lisk-core transaction:send 0802100018022080c2d72f2a20e03c09bdc8c023d94cf66a5d352e6258380210d97d545abbf75668ea3736e3123229088094ebdc031214ab0041a7d3f7b2c290b5b834d46bdc7b7eb858151a0b73656e6420746f6b656e733a40faa2626d7306506b1999f48aa2f4b1ffdee01e641fa76d37a9d1d6fd8c225a81065c856ea625c52d138a7e3ba86b62913dc8e5aef8b5e307641ab66e0277a60b
----

==== Via the HTTP API for Lisk nodes

> If the xref:{url_sdk_httpapi}[] is enabled on a node, it is possible to send the transaction to the node via HTTP request:

Reviewer: ```suggestion
If the xref:{url_sdk_httpapi}[] is enabled on a node, it is possible to send the transaction to the node via an HTTP request:
```
